File: Perl/cursor_position.pl
#
#   Checks if cursor is in same position after sleeping N seconds (default: 20 min), Perl
#   Module written by Brandon Arvanaghi 
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

use Win32::API;

my $secs = 1200;
if (@ARGV == 1) {
	$secs = $ARGV[0];
}

Win32::API::Struct->typedef( POINT => qw{
    LONG x; 
    LONG y; 
});

Win32::API->Import('user32', 'BOOL GetCursorPos(LPPOINT lpPoint)');

my $point = Win32::API::Struct->new('POINT');
$point->align('auto');
$point->{'x'} = 0; $point->{'y'} = 0;
GetCursorPos($point);
print "x: $point->{x}, y: $point->{y}\n";

sleep($secs);

my $point2 = Win32::API::Struct->new('POINT');
$point2->{'x'} = 0; $point2->{'y'} = 0;
GetCursorPos($point2);
print "x: $point2->{x}, y: $point2->{y}\n";

if ($point->{x} - $point2->{x} == 0 && $point->{y} - $point2->{y} == 0) {
	print "The cursor has not moved in the last $secs seconds. Do not proceed.\n"
} else {
	print "The cursor is not in the same position as it was $secs seconds ago. Proceed!\n"
}

File: Python/cursor_position.py
#
#   Checks if cursor is in same position after sleeping N seconds (default: 20 min), Python
#   Module written by Brandon Arvanaghi
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

from time import sleep
import win32api
import sys

secs = 1200;
if len(sys.argv) == 2:
	secs = float(sys.argv[1])

x, y = win32api.GetCursorPos()
print("x: " + str(x) + ", y: " + str(y))

sleep(secs)

x2, y2 = win32api.GetCursorPos()
print("x: " + str(x2) + ", y: " + str(y2))

if x - x2 == 0 and y - y2 == 0:
	print("The cursor has not moved in the last {} seconds. Do not proceed.".format(secs))
else:
	print("The cursor is not in the same position as it was {} seconds ago. Proceed!".format(secs))
